src/common/ipc/ipc.ts

```diff
@@ -30,7 +30,15 @@ import { ClusterFrameInfo, clusterFrameMap } from "../cluster-frames";
 import type { Disposer } from "../utils";
 import type remote from "@electron/remote";
 
-const electronRemote = ipcMain ? null : require("@electron/remote");
+const electronRemote = (() => {
+  if (ipcRenderer) {
```

Comment:
This is non-env-agnostic code in common, which is bad. However, the tooling to solve something like this is not here yet -> ok for now.